contents/docs/plugins/build/overview.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Overview
+sidebar: Docs
+showTitle: true
+---
+
+A great way to understand what you can do with plugins is to understand how data flows throw plugins.
+
+There's 2 big concepts to remember:
+
+1. Every plugin acts on a single event coming in.
+
+2. Plugins act on events before they are stored.
+
+Each plugin can add more information to the event, or even modify existing properties. For example, the [GeoIP plugin](https://posthog.com/plugins/geoip) is a plugin that adds GeoIP information to the event.
+
+Note that the output of one plugin goes into the next plugin. Here's how this looks:
+
+![GeoIP Plugin Example](../../../images/plugins/geoip-plugin-example.png)
+
+However, this isn't all. Plugins _don't_ have to modify events at all: they can do other things when an event comes in.
+
+For example, you can send an event to AWS S3 whenever you see it in PostHog. Indeed, the [S3 plugin](https://posthog.com/plugins/s3-export) does exactly that. In this case, it doesn't matter if the S3 export succeeds or not, the event will always be stored.
+
+![S3 Plugin Example](../../../images/plugins/s3-plugin-example.png)
```

Comment:
this image is a bit wrong, as it gives me `processEvent` vibes. I'd personally prefer 2 arrows coming out of the event, one going via the plugin to S3, and the other to the DB

Reply:
Hmmmm, as a reader I feel that to be more confusing: how/why did the event go to both places?

I think if I remove the second event from the diagram, it feels closer to `onEvent` ?

Reply:
![image](https://user-images.githubusercontent.com/7115141/122248170-cbe83400-cebf-11eb-864e-116776d37bd4.png)

I really do think this raises more questions than it answers.

Disregarding how things actually happen, a good model helps people reason about the system without knowing exactly what's happening in the system. And **I think** all people need to know is that it's like `processEvent`, but the return value doesn't matter, so they can do all sorts of "side stuff" with `onEvent`.